supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/api/PipelinesApiRequestManager.scala

```diff
@@ -328,7 +347,15 @@ object PipelinesApiRequestManager {
     override def getMessage: String = e.getMessage
   }
 
-  class PAPIApiException(val e: Throwable) extends RuntimeException(e) with CromwellFatalExceptionMarker {
-    override def getMessage: String = "Unable to complete PAPI request"
+  sealed trait PAPIApiException extends RuntimeException with CromwellFatalExceptionMarker {
+    def e: Throwable
+  }
+
+  class SystemPAPIApiException(val e: Throwable) extends PAPIApiException {
```

Comment:
This new constructor stack (scala-class extending a scala-trait that extends a java-class) breaks the stack trace (though not worse than we already do with `Validated[String, A]`). Ideally the wrapped ~`cause`~ `e` should be retained for debugging purposes if/when this `Exception` ever manifests.

Reply:
I think it's the wrapping of a `Throwable` inside a `class` (which also happens to be an `Exception`) is what's losing the stack trace - and if so I can carry-forward that stack trace by adding a `override def cause = ...` method here.

Having said that, I've been a little bit on a `with NoStackTrace` drive recently for a bunch of these "Exceptions as signals" that we seem to be passing around but never actually throwing.

Reply:
One could always cease passing exceptions around as signals :)

Reply:
The "cease doing" part is easy enough... it's the "undo all previous instances" which is trickier